ravenframework/Optimizers/GeneticAlgorithm.py

```diff
@@ -669,6 +1086,76 @@ def _resolveNewGeneration(self, traj, rlz, objectiveVal, fitness, g, info):
     else: # e.g. rerun
       pass # nothing to do, just keep moving
 
+  def _resolveNewGenerationMulti(self, traj, rlz, info):
```

Comment:
Could you checkout this function and `_resolveNewGeneration` to see if you can combine them together?